crates/wash-cli/src/dev.rs

```diff
@@ -152,25 +319,482 @@ impl KnownDep {
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+enum VersionCoverage {
+    #[default]
+    All,
+    SemVer(Version),
+}
+
+/// Specification of a dependency (possibly implied)
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+struct DependencySpecInner {
+    /// Relevant WIT information that represents the dependency
+    ///
+    /// The interfaces that the dependency receives
+    ///
+    /// This generally means that the component will
+    wit: (WitNamespace, WitPackage, WitInterface, VersionCoverage),
+
+    /// Whether this dependency should delegated to the workspace
+    delegated_to_workspace: bool,
+
+    /// Image reference to the component that should be inserted/used
+    ///
+    /// This reference *can* be missing if an override is specified with no image,
+    /// which can happen in at least two cases:
+    /// - custom WIT-defined interface imports/exports (we may not know at WIT processing what their overrides will be)
+    /// - project-level with workspace delegation  (we may not know what the image ref is at the project level)
+    image_ref: Option<String>,
+
+    /// Whether this dependency represents a WebAssembly component, rather than a (standalone binary) provider
+    is_component: bool,
+
+    /// The link name this dependency should be connected over
+    ///
+    /// In the vast majority of cases, this will be "default", but it may not be
+    /// if the same interface is used to link to multiple different providers/components
+    link_name: LinkName,
+
+    /// Configurations that must be created and/or consumed by this dependency
+    configs: Vec<wadm_types::ConfigProperty>,
+
+    /// Secrets that must be created and/or consumed by this dependency
+    ///
+    /// [`SecretProperty`] here support a special `policy` value which is 'env'.
+    /// Paired with a key that looks like "$SOME_VALUE", the value will be extracted from ENV *prior* and
+    secrets: Vec<wadm_types::SecretProperty>,
+}
+
+impl DependencySpecInner {
+    /// Retrieve the name for this dependency
+    fn name(&self) -> String {
+        match self.image_ref.as_deref() {
+            Some(DEFAULT_KEYVALUE_PROVIDER_IMAGE) => "keyvalue-nats".into(),
+            Some(DEFAULT_HTTP_CLIENT_PROVIDER_IMAGE) => "http-client".into(),
+            Some(DEFAULT_HTTP_SERVER_PROVIDER_IMAGE) => "http-server".into(),
+            Some(DEFAULT_BLOBSTORE_FS_PROVIDER_IMAGE) => "blobstore-fs".into(),
+            Some(DEFAULT_MESSAGING_NATS_PROVIDER_IMAGE) => "messaging-nats".into(),
+            _ => format!("custom-{}-{}", self.wit.0, self.wit.1),
         }
     }
+
+    /// Retrieve the image reference for this dependency
+    fn image_ref(&self) -> Option<&str> {
+        self.image_ref.as_deref()
+    }
+
+    /// Retrieve whether this dependency spec is a component
+    fn is_component(&self) -> bool {
+        self.is_component
+    }
+}
+
+// #[derive(Debug, Clone, PartialEq, Eq)]
+// enum SecretSpec {
```

Comment:
nit: can be removed if not needed but disregard if you're planning to use this in a followup

Reply:
Oh thanks for catching this, need to remove